### PR TITLE
Symmetrize

### PR DESCRIPTION
**Pull request by DavidAKopriva**
Add capability to take a mesh and reflect it about a (straight line) boundary with the name "symmetry". This should answer the request of some users for a symmetric mesh. TODO: Add example, testing, documentation, and test the mesh files.

### CODE REVIEW

**Comment by DavidAKopriva** [comment]
I've updated everything I can think of. One thing especially open for discussion is that I append "_R" to a boundary name on the reflected portion of the mesh. I guess if one does not want that, it's a quick search/replace (maybe) in the file. Maybe something else truly unique would be better ? "_#R?

**Review by andrewwinters5000 — CHANGES_REQUESTED** [review]
Looks good overall; just a small suggestion with the literals.

One follow-up question. If one sets multiple symmetry boundary it only reflects the "base" mesh, right? It would not be the case that the base mesh gets reflected over one boundary and the entire mesh is then reflected over the second boundary?

File: Source/Foundation/LineReflectionModule.f90
Comment (DavidAKopriva) [reply Source/Foundation/LineReflectionModule.f90:202]: Absolutely. Can't believe I did that. Always type literals is the rule.

Comment (DavidAKopriva) [reply Source/Foundation/LineReflectionModule.f90:202]: Not sure what the followup question is. There can only be reflection about a single *line*. If multiple boundaries are designated as 'symmetry' lines then they must therefore be co-linear. If multiple boundaries are designated as symmetry and they are not along the same line, then an error exception is thrown saying it can't do that. Is this the case that you are thinking of?

Comment (andrewwinters5000) [reply Source/Foundation/LineReflectionModule.f90]: Ah, maybe I misunderstood. So I cannot put, say, the left boundary and the right boundary as symmetry and then reflect the mesh over each. I am only allowed to have, say, the left boundary set to symmetry and this left boundary could be composed as a chain of colinear line segments. The latter makes more sense.

Comment (DavidAKopriva) [reply Source/Foundation/LineReflectionModule.f90]: Right. I would say the first situation has two *reflection* boundaries. The second, which is what I have implemented is to have a symmetry boundary, which is a single line across which one will reflect. What you describe does not generate a symmetric mesh, I don't think. (Nope, it doesn't... I can sketch out an easy one with a graded mesh in a box.) There is nothing wrong with what you describe, but that's not implemented. Another option is to not reflect but to reproduce and shift a mesh to make a repeated pattern. The way I implemented the symmetric mesh can be extended to both of the cases (reflection and reproduction), though how to write a logical command for it would take some thought. Right now, the key is to have collinear boundaries (think of a semi-annular region with two straight lines at the ends) and reflect about that. Does all that make sense?

Comment (andrewwinters5000) [reply Source/Foundation/LineReflectionModule.f90]: Yes, that makes sense. The requested feature was that one generates half of the mesh and then reflects it to create the other half. The situation I envisioned would likely never come up because, with some "pre-processing," one could define an appropriate partial domain to mesh that then requires a single reflection. 

A natural extension of this feature would be reflecting over a `CircularArc` or `Spline`, but that should be the topic of a different PR.

**Review by andrewwinters5000 — APPROVED** [review]
LGTM! Once this is merged and released I need to add the capability to the HOHQMesh.jl counterpart and add some test and a tutorial. (This is mostly a written note to myself more than anything, you do not need to do anything.)

File: Source/Foundation/LineReflectionModule.f90
Comment (DavidAKopriva) [reply Source/Foundation/LineReflectionModule.f90]: Hmm, like in complex variables with a 1/r transformation? Yeah, a different PR!

**Comment by DavidAKopriva** [comment]
No, there's nothing that you have to do. Just call at least one of the boundary curves "symmetry" when you are generating a model and off it will go. (Hey, this might be a way to define an interior boundary! If one of the curves along a line is called symmetry and the others aren't, then the nodes will be duplicated, along those others and the curve names are not deleted. Hmm.)

**Comment by andrewwinters5000** [comment]
Reflection

**Comment by andrewwinters5000** [comment]
Reflection

**Pull request merged**
